pkg/agent/client/client.go

```diff
@@ -90,7 +93,11 @@ func (t *grpcTunnel) serve() {
 		switch pkt.Type {
 		case agent.PacketType_DIAL_RSP:
 			resp := pkt.GetDialResponse()
-			if ch, ok := t.pendingDial[resp.Random]; !ok {
+			t.pendingDialLock.RLock()
+			ch, ok := t.pendingDial[resp.Random]
+			t.pendingDialLock.RUnlock()
```

Comment:
Please use defer for the unlock. If we need to build a helper to minimize how long the lock is head, that's fine.

Reply:
@cheftako the lock is in a loop so it's not `defer`able. it could be a method to get the map, but it's just a couple of occurrences. If getting the map becomes repeated pattern, let's put it in a method.

Reply:
You could either write a getter method or an anonymous method to allow you to defer here. If something happened, such that we errored here and failed to unlock the lock, this would become very hard to debug. I would rather we just add belt and suspenders to protect ourselves.

Reply:
it's a map lookup, which shouldn't error out. but i am fine either way.

Reply:
I figured it is a little bit too verbose to add 3 methods for every map that is protected. golang doesn't error out like an exception is thrown in other languages, so as long as we don't `return` or `break` in between, it's fine.